### ISSUE TITLE
Push plugin does not work on Android 24

### ISSUE THREAD

**Issue by vchimev** [NativeScript/push-plugin#40]
Hi,

NativeScript CLI 2.3.0 is going to support Android Build Tools 24.

Currently, if you try to run the [sample](https://github.com/NativeScript/sample-push-plugin) app with this setup it will crash:

```
08-29 10:03:51.140  3557  3581 E AndroidRuntime: FATAL EXCEPTION: Thread-193
08-29 10:03:51.140  3557  3581 E AndroidRuntime: Process: org.nativescript.allsdk, PID: 3557
08-29 10:03:51.140  3557  3581 E AndroidRuntime: java.lang.IncompatibleClassChangeError: The method 'java.io.File android.support.v4.content.ContextCompat.getNoBackupFilesDir(android.content.Context)' was expected to be of type virtual but instead was found to be of type direct (declaration of 'com.google.android.gms.iid.zzd' appears in /data/app/org.nativescript.allsdk-1/base.apk)
08-29 10:03:51.140  3557  3581 E AndroidRuntime: 	at com.google.android.gms.iid.zzd.zzeb(Unknown Source)
08-29 10:03:51.140  3557  3581 E AndroidRuntime: 	at com.google.android.gms.iid.zzd.<init>(Unknown Source)
08-29 10:03:51.140  3557  3581 E AndroidRuntime: 	at com.google.android.gms.iid.zzd.<init>(Unknown Source)
08-29 10:03:51.140  3557  3581 E AndroidRuntime: 	at com.google.android.gms.iid.InstanceID.zza(Unknown Source)
08-29 10:03:51.140  3557  3581 E AndroidRuntime: 	at com.google.android.gms.iid.InstanceID.getInstance(Unknown Source)
08-29 10:03:51.140  3557  3581 E AndroidRuntime: 	at com.telerik.pushplugin.ObtainTokenThread.getTokenFromGCM(ObtainTokenThread.java:39)
08-29 10:03:51.140  3557  3581 E AndroidRuntime: 	at com.telerik.pushplugin.ObtainTokenThread.run(ObtainTokenThread.java:31)
08-29 10:03:51.143  1534  2673 W ActivityManager:   Force finishing activity org.nativescript.allsdk/com.tns.NativeScriptActivity
08-29 10:03:51.152  1193  1193 D gralloc : Registering a buffer in the process that created it. This may cause memory ordering problems.
```

Possible solution - Update Android Support Library from v4 to v7:
https://github.com/NativeScript/push-plugin/tree/master/native-src/android/libs

CC: @AntonDobrev.

**Comment by bnussey** [comment]
Any updates on this @vchimev and @AntonDobrev ? Experiencing this issue myself.

**Comment by AntonDobrev** [comment]
@bnussey A workaround for now is to uninstall the Android Build Tools for level 24 and use a lower version of them.

**Comment by svzi** [comment]
@AntonDobrev Nothing else we could do about?

I've had removed the build tools for level 24, but that doesn't make any difference at all. The sample app still crashes with the error message. This is what I have installed from the Android SDK:
![bildschirmfoto 2016-09-28 um 07 38 18](https://cloud.githubusercontent.com/assets/1890513/18901843/a7a48e6e-854e-11e6-864a-5d92ae737294.png)

My TNS configuration:
`┌──────────────────┬─────────────────┬────────────────┬───────────────┐
│ Component        │ Current version │ Latest version │ Information   │
│ nativescript     │ 2.3.0           │ 2.3.0          │ Up to date    │
│ tns-core-modules │ 2.3.0           │ 2.3.0          │ Up to date    │
│ tns-android      │ 2.3.0           │ 2.3.0          │ Up to date    │
│ tns-ios          │                 │ 2.3.0          │ Not installed │
└──────────────────┴─────────────────┴────────────────┴───────────────┘`

I really need to add this plugin, but I don't want to mess around with my development environment in order to get it running. What about the possible solution (updating the Android Support Library from v4 to v7) mentioned by @vchimev ?

**Comment by AntonDobrev** [comment]
@svzi I believe you need to remove the android platform from the NS project and add it again after the change. You are correct, the workaround is not convenient. I escalated the issue to the plugin collaborators so stay tuned.

**Comment by svzi** [comment]
@AntonDobrev Damn it, I was so sure that I did that as well, but I just did it again and at least the sample app is working. Thanks for that, I will take some more looks now.

**Comment by manojdcoder** [comment]
Is error below related to this ticket?

```
java.lang.IncompatibleClassChangeError: The method 'java.io.File android.support.v4.content.ContextCompat.getNoBackupFilesDir(android.content.Context)' was expected to be of type virtual but instead was found to be of type direct (declaration of 'com.google.android.gms.iid.zzd' appears in /data/app/com.SlateUp.Plunk-1/base.apk)
at com.google.android.gms.iid.zzd.zzeb(Unknown Source)
at com.google.android.gms.iid.zzd.<init>(Unknown Source)
at com.google.android.gms.iid.zzd.<init>(Unknown Source)
at com.google.android.gms.iid.InstanceID.zza(Unknown Source)
at com.google.android.gms.iid.InstanceID.getInstance(Unknown Source)
at com.telerik.pushplugin.ObtainTokenThread.getTokenFromGCM(ObtainTokenThread.java:39)
at com.telerik.pushplugin.ObtainTokenThread.run(ObtainTokenThread.java:31)
```

**Comment by yairvillarp** [comment]
Hi, I'm having the same problem, any update on this ticket??

**Comment by manojdcoder** [comment]
@yairvillarp Uninstalling build tools 24 didn't work for me, I did some patches here https://github.com/manojdcoder/push-plugin made it working. You may give it a try.

**Comment by AntonDobrev** [comment]
Fixed in 0.0.17  just released. Thanks to @manojdcoder for the provided ideas. 

If needed:
- pull the npm package
- in your project: `tns platform remove android` and then `tns platform add android`
- build
